src/NuGet.Core/NuGet.Versioning/VersionRange.cs

```diff
@@ -344,7 +344,7 @@ private static NuGetVersion GetNonSnapshotVersion(NuGetVersion version)
                 }
 
                 // Remove the last label and add in the fixed label if one exists.
-                var fixedLabels = version.ReleaseLabels.Take(version.ReleaseLabels.Count() - 1).ToList();
+                var fixedLabels = version.ReleaseLabels.Take(version.ReleaseLabels.Count() - 1).AsList();
```

Comment:
Same

Reply:
You should probably review all changes from ToList() to AsList() to ensure we only do that when the underlying collection is already a List.